src/mpc-hc/MiniDump.cpp

```diff
@@ -0,0 +1,196 @@
+#include "stdafx.h"
+#include "MiniDump.h"
+#include <windows.h>
+#include <psapi.h>
+
+#ifndef _DEBUG
+
+LPCWSTR GetExceptionName(DWORD code)
+{
+    switch (code) {
+        case EXCEPTION_ACCESS_VIOLATION:
+            return _T("ACCESS VIOLATION");
+        case EXCEPTION_ARRAY_BOUNDS_EXCEEDED:
+            return _T("ARRAY BOUNDS EXCEEDED");
+        case EXCEPTION_BREAKPOINT:
+            return _T("BREAKPOINT");
+        case EXCEPTION_DATATYPE_MISALIGNMENT:
+            return _T("DATATYPE MISALIGNMENT");
+        case EXCEPTION_FLT_DENORMAL_OPERAND:
+            return _T("FLT DENORMAL OPERAND");
+        case EXCEPTION_FLT_DIVIDE_BY_ZERO:
+            return _T("FLT DIVIDE BY ZERO");
+        case EXCEPTION_FLT_INEXACT_RESULT:
+            return _T("FLT INEXACT RESULT");
+        case EXCEPTION_FLT_INVALID_OPERATION:
+            return _T("FLT INVALID OPERATION");
+        case EXCEPTION_FLT_OVERFLOW:
+            return _T("FLT OVERFLOW");
+        case EXCEPTION_FLT_STACK_CHECK:
+            return _T("FLT STACK CHECK");
+        case EXCEPTION_FLT_UNDERFLOW:
+            return _T("FLT UNDERFLOW");
+        case EXCEPTION_GUARD_PAGE:
+            return _T("GUARD PAGE");
+        case EXCEPTION_ILLEGAL_INSTRUCTION:
+            return _T("ILLEGAL_INSTRUCTION");
+        case EXCEPTION_IN_PAGE_ERROR:
+            return _T("IN PAGE ERROR");
+        case EXCEPTION_INT_DIVIDE_BY_ZERO:
+            return _T("INT DIVIDE BY ZERO");
+        case EXCEPTION_INT_OVERFLOW:
+            return _T("INT OVERFLOW");
+        case EXCEPTION_INVALID_DISPOSITION:
+            return _T("INVALID DISPOSITION");
+        case EXCEPTION_INVALID_HANDLE:
+            return _T("INVALID HANDLE");
+        case EXCEPTION_NONCONTINUABLE_EXCEPTION:
+            return _T("NONCONTINUABLE EXCEPTION");
+        case EXCEPTION_PRIV_INSTRUCTION:
+            return _T("PRIV INSTRUCTION");
+        case EXCEPTION_SINGLE_STEP:
+            return _T("SINGLE STEP");
+        case EXCEPTION_STACK_OVERFLOW:
+            return _T("STACK OVERFLOW");
+        case 0xE06D7363:
+            return _T("C++ EXCEPTION");
+        default:
+            return _T("[Unknown]");
+    }
+}
+
+HMODULE GetExceptionModule(LPVOID address, LPWSTR moduleName)
+{
+    HMODULE moduleList[1024];
+    DWORD sizeNeeded = 0;
+    if (FALSE == EnumProcessModules(GetCurrentProcess(), moduleList, 1024, &sizeNeeded) || sizeNeeded < sizeof(HMODULE)) {

```

Comment:
Generally IMO we should use the helper macros like `_countof` instead of using hardcoded numbers.